Add docu for chat suggestions (#788)
eclipse-theia/theia-website: src/docs/theia_ai.md

Custom change set element implementations are identified by a `uri`, have full control over their presentation (label, icon, additional information), and can specify whether and how actions, such as open, open change, accept and discard are implemented. See [`ChangeSetElement` interface](https://github.com/eclipse-theia/theia/blob/451464e6ea3d4aaf9cdbffd3d17dbb117787fc4e/packages/ai-chat/src/common/chat-model.ts#L100) for more details.

## Chat Suggestions

Theia AI provides a mechanism for displaying contextual suggestions to users in the chat interface. These suggestions can help guide users on what to do next, offer shortcuts to common actions, or provide contextual help based on the current state of the conversation.

The following video demonstrates how suggestions look like in the UI.

<video controls>
  <source src="../../agent-suggestions.webm" type="video/webm">
  Your browser does not support the video tag.
</video>

> You can add suggestions to a chat session by accessing the chat model API, e.g. from within an agent implementation. The suggestions can text augmented with different types of actions:

Reviewer: ```suggestion
You can add suggestions to a chat session by accessing the chat model API, e.g. from within an agent implementation. The suggestions can be text augmented with different types of actions:
```

1. **Command-based suggestions**: Trigger Theia commands when clicked

> 2. **Callback-based suggestions**: Execute custom code (a call back) when clicked

Reviewer: ```suggestion
2. **Callback-based suggestions**: Execute custom code (a callback) when clicked
```

> Here's an example of how to add suggestions to a chat model with a simple call back:

Reviewer: ```suggestion
Here's an example of how to add suggestions to a chat model with custom callback code when a link in the suggestion is clicked:
```

```typescript
model.setSuggestions([
    {
        kind: 'callback',
        callback: () => this.chatService.sendRequest(session.id, { text: '@Coder please look at src/docs/theia_ai.md and fix any problems.' }),
        content: '[Fix problems](_callback) in the current file.'
    },
]);
```

In this example, when the user clicks on the "Fix problems" suggestion, it will automatically send a request to the Coder agent to analyze and fix issues in the specified file.

> For command-based suggestions, you can use a MarkDownString kind and specify a command ID it:

Reviewer: ```suggestion
For command-based suggestions, you can use `MarkdownStringImpl` and specify command identifier in links:
```
